tests/models/StakeRequestRepository/util.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+import {
+  StakeRequestAttributes,
+  StakeRequest,
+} from '../../../src/models/StakeRequestRepository';
+
+import assert = require('assert');
+
+const Util = {
+  checkStakeRequestAgainstAttributes(
+    stakeRequest: StakeRequest,
+    stakeRequestAttributes: StakeRequestAttributes,
+  ): void {
+    assert.strictEqual(
+      stakeRequest.stakeRequestHash,
+      stakeRequestAttributes.stakeRequestHash,
+    );
+
+    assert.strictEqual(
+      stakeRequest.messageHash,
+      stakeRequestAttributes.messageHash,
+    );
+
+    assert.strictEqual(
+      stakeRequest.amount,
+      stakeRequestAttributes.amount,
+    );
+
+    assert.strictEqual(
+      stakeRequest.beneficiary,
+      stakeRequestAttributes.beneficiary,
+    );
+
+    assert.strictEqual(
+      stakeRequest.gasPrice,
+      stakeRequestAttributes.gasPrice,
+    );
+
+    assert.strictEqual(
+      stakeRequest.gasLimit,
+      stakeRequestAttributes.gasLimit,
+    );
+
+    assert.strictEqual(
+      stakeRequest.nonce,
+      stakeRequestAttributes.nonce,
+    );
+
+    assert.strictEqual(
+      stakeRequest.gateway,
+      stakeRequestAttributes.gateway,
+    );
+
+    assert.strictEqual(
+      stakeRequest.stakerProxy,
+      stakeRequestAttributes.stakerProxy,
+    );
```

Comment:
For all the asserts above,  `reason` text,  the third param is missing.
We have followed this in all tests.

Reply:
I would propose rather to put a reason test for asserts in test (I am speaking about tests here only), that are not obvious from the test context. In this exact case, from the function name it's already obvious what are those assertions. What are your thoughts?

Reply:
If anything fails in `checkStakeRequestAgainstAttributes`, it will be difficult to find exactly what has failed just by seeing the test logs.
We may know that by seeing the line number, but here we can add the exact reason as we know what we are comparing.

Reply:
You are right: a stack trace will show us the exact position where it has failed.

I still do not think, that adding a reason string to something that is already obvious going to contribute something useful. Function says `checkStakeRequestAgainstAttributes` by passing stakeRequest and stakeRequestAttributes and then simply comparing member by member. Adding `Stake request's staker proxy does not match with stake request attributes' staker proxy` is verbose and does not contribute something useful.

Reply:
When I will run all the tests, I will not know if the broken test is related to `checkStakeRequestAgainstAttributes`.  Our practice was to always put the reason.

Reply:
@deepesh-kn, sorry I am not sure I follow you.

I have changed the Utils::checkStakeRequestAgainstAttributes 

```ts
    assert.strictEqual(
      stakeRequest.stakerProxy,
      `${stakeRequestAttributes.stakerProxy}1`,
    );
```

here is the output when I run test:

```bash
- 'stakerProxy'
+ 'stakerProxy1'
      + expected - actual

      -stakerProxy
      +stakerProxy1
      
      at Object.checkStakeRequestAgainstAttributes (tests/models/StakeRequestRepository/util.ts:69:12)
      at Object.<anonymous> (tests/models/StakeRequestRepository/updateMessageHash.test.ts:83:10)
      at Generator.next (<anonymous>)
      at fulfilled (tests/models/StakeRequestRepository/updateMessageHash.test.ts:19:58)
```

or you've meant something else?

Reply:
We can take this discussion offline, I think this should not block the PR from merging.